daisy/step_deprecate_images.go

```diff
@@ -51,7 +54,11 @@ func (d *DeprecateImages) validate(ctx context.Context, s *Step) DError {
 			return Errf("cannot deprecate image %q: project does not exist: %q", di.Image, di.Project)
 		}
 
-		if !strIn(di.DeprecationStatus.State, deprecationStates) {
+		// Verify State is one of the deprecated states.
+		// The Alpha check also requires the value to not be emptry string as in that case the GA API will be used.
+		if di.DeprecationStatusAlpha.State != "" && strIn(di.DeprecationStatusAlpha.State, deprecationStates) {
```

Comment:
I forgot, can di.DeprecationStatusAlpha be null?

Reply:
The deprecationStates lists empty string as an allowed value, so requiring di.DeprecationStatusAlpha.State to not be empty string ensures that when it's not set in the Alpha API data structure, we use the GA data structure.